***INTEGRANTES: Daniel Esteban Olmos - Juan Pablo Gonzalez - Juan Rafael Ortiz***

# TALLER 2do corte 

## Instale la librería wooldridge e importe numpy y pandas.

In [ ]:
!pip install wooldridge
import pandas as pd
import numpy as np
import wooldridge as wd

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 5.1/5.1 MB 39.5 MB/s eta 0:00:00


## Cargue el dataset wage1 y guárdelo en una variable. Conserve únicamente las columnas exper, educ y wage.

In [ ]:
df=wd.data("wage1") [['exper', 'educ', 'wage']]

In [ ]:
df

exper  educ   wage
0        2    11   3.10
1       22    12   3.24
2        2    11   3.00
3       44     8   6.00
4        7    12   5.30
..     ...   ...    ...
521     14    16  15.00
522      2    10   2.27
523     13    15   4.67
524      5    16  11.56
525      5    14   3.50

[526 rows x 3 columns]

##Pase el dataframe a un array de numpy.

In [ ]:
y= df["wage"]
x1= df[["educ","exper"]]


In [ ]:
array=np.array(x1)

In [ ]:
array

array([[11,  2],
       [12, 22],
       [11,  2],
       ...,
       [15, 13],
       [16,  5],
       [14,  5]])

In [ ]:
x=array

In [ ]:
p= np.ones(len(x))

In [ ]:
p

array([1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1., 1.,
       1., 1., 1., 1., 1.

In [ ]:
X=np.c_[p,x]

In [ ]:
X.shape

(526, 3)

In [ ]:
X

array([[ 1., 11.,  2.],
       [ 1., 12., 22.],
       [ 1., 11.,  2.],
       ...,
       [ 1., 15., 13.],
       [ 1., 16.,  5.],
       [ 1., 14.,  5.]])

TRANSPUESTA

In [ ]:
k= X.T.dot(X)

In [ ]:
k

array([[   526.,   6608.,   8951.],
       [  6608.,  87040., 106539.],
       [  8951., 106539., 249027.]])

INVERSA DE TRANSPUESTA

In [ ]:
inv_k= np.linalg.inv(k)

In [ ]:
inv_k

array([[ 5.53926242e-02, -3.71227943e-03, -4.02835200e-04],
       [-3.71227943e-03,  2.72907389e-04,  1.66782433e-05],
       [-4.02835200e-04,  1.66782433e-05,  1.13597903e-05]])

In [ ]:
x_y=X.T.dot(y)

In [ ]:
betas=inv_k.dot(x_y)

**1. BETAS**

In [ ]:
betas

array([-3.39053946,  0.64427208,  0.0700954 ])

In [ ]:
y_estimada= X.dot(betas)

In [ ]:
betas

array([-3.39053946,  0.64427208,  0.0700954 ])

In [ ]:
u=y-y_estimada

In [ ]:
sigma= sum(u**2)/523

In [ ]:
sigma

10.608336116213543

In [ ]:
l=sigma*inv_k


**2. ERRORES ESTANDAR**

In [ ]:
ee= np.diag(np.sqrt(l))

<ipython-input-34-e0c3c7e6d8f3>:1: RuntimeWarning: invalid value encountered in sqrt
  ee= np.diag(np.sqrt(l))


In [ ]:
ee

array([0.76656609, 0.05380607, 0.01097764])

**3. ESTADISTICO T**

In [ ]:
t=betas/ee

In [ ]:
t

array([-4.423023  , 11.97396614,  6.38529107])

**4. iNTERVALOS DE CONFIANZA**


Pimer coeficiente

In [132]:
interv_lower=betas-(1.96*ee)



In [133]:
interv_lower

array([-4.89300901,  0.53881218,  0.04857923])

In [134]:
interv_upper=betas+(1.96*ee)

In [135]:
interv_upper

array([-1.88806992,  0.74973198,  0.09161156])

In [137]:
intervalos= (interv_lower,interv_upper)

In [138]:
intervalos

(array([-4.89300901,  0.53881218,  0.04857923]),
 array([-1.88806992,  0.74973198,  0.09161156]))

COMPROBACION

In [ ]:
y=df["wage"]
x=df[["educ","exper"]]
X= sm.add_constant(x)
model= sm.OLS(y,X)
reg=model.fit()
reg.summary()

import statsmodels.api as sm

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                   wage   R-squared:                       0.225
Model:                            OLS   Adj. R-squared:                  0.222
Method:                 Least Squares   F-statistic:                     75.99
Date:                Fri, 21 Apr 2023   Prob (F-statistic):           1.07e-29
Time:                        19:37:47   Log-Likelihood:                -1366.0
No. Observations:                 526   AIC:                             2738.
Df Residuals:                     523   BIC:                             2751.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         -3.3905      0.767     -4.423      0.000      -4.896      -1.885
educ           0.6443      0.054     11.974      0.000       0.539       0.750
exper          0.0701      0.011      6.385      0.000       0.049       0.092
==============================================================================
Omnibus:                      196.872   Durbin-Watson:                   1.820
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              727.160
Skew:                           1.715   Prob(JB):                    1.26e-158
Kurtosis:                       7.627   Cond. No.                         130.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

**5 INTERPRETACION **

**El coeficiente de educación (0.6443) indica que:**  Por cada aumento de  1 unidad  en la variable de educación, el salario  aumenta en 0.6443 unidades. 

**El coeficiente de experiencia (0.0701) indica que:**  por cada aumento de 1 unidad en  la variable de experiencia, el salario  aumenta en 0.0701 unidades, manteniendo constante la variable de educación.

Para saber la significancia estadistica podemos determinarla mediante los valores de t y los intervaloes de confianza del 95% de cada uno de los valores. 

**t:** Si el valor de t es mayor a 1.96 , teniendo como base una prueba de dos colas, se puede determinar que la variable es significativa. En ese orden de ideas:

**variable educ:** 11.97>1.96  , esto nos indica que la variable educ es significativa en relacion a la variable wage 

**variable exp:** 6.385 >1.96  , esto nos indica que la variable exp es significativa en relacion a la variable wage


